Add a public doc for passkeys for testnet launch

abhinavg6 opened the pull request:
## Description 

Adding a public doc for passkeys for testnet launch. It's a high-level doc that links to the relevant to the TS SDK doc and the SIP.

## Test plan 

@joyqvq has done the relevant dev testing, and beyond that we've tested the feature in devnet using a sample app and a community-built app.

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] gRPC:
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

abhinavg6 commented:
cc @joyqvq @ronny-mysten for review - thank you

joyqvq commented on docs/content/concepts/cryptography.mdx:
```diff
@@ -14,10 +14,16 @@ Go to [Transaction Authentication](./cryptography/transaction-auth.mdx).
 
+## Passkeys
+
+Passkeys is a Sui primitive that enables you to sign-in to apps and send transactions to Sui using a private key securely stored on a authenticator. It uses the WebAuthn standard.
```
```suggestion
Sui supports passkey signatures that enables you to sign-in to apps and sign transactions to Sui using a private key securely stored on a passkey authenticator. It uses the WebAuthn standard.
```

joyqvq commented on docs/content/concepts/cryptography/passkeys.mdx:
```diff
@@ -0,0 +1,60 @@
+---
+title: Passkeys
+description: Passkeys is a Sui primitive that enables you to sign-in to apps and send transactions to Sui using a private key securely stored on a authenticator. It uses the WebAuthn standard.
+---
```
same as 
```
Sui supports passkey signature scheme that enables you to sign-in to apps and sign transactions to Sui using a private key securely stored on a passkey authenticator. It uses the WebAuthn standard.
```

(since passkey itself is not a sui primitive, sui just has ability to verify the signature produced by a passkey on a transaction now)

joyqvq commented on docs/content/concepts/cryptography/passkeys.mdx:
```diff
@@ -0,0 +1,60 @@
+
+Passkeys provide a secure and user-friendly alternative for submitting transactions to Sui. Built on the **WebAuthn standard**, passkeys let users authenticate and sign transactions using:
+
+- Hardware security keys, such as YubiKeys)
```
```suggestion
- Hardware security keys, such as YubiKeys
```

joyqvq commented on docs/content/concepts/cryptography/passkeys.mdx:
```diff
@@ -0,0 +1,60 @@
+
+Passkeys simplify authentication by removing the need to manage seed phrases or private keys manually. Instead, they rely on device-based authentication and cloud synchronization, allowing seamless, phishing-resistant access across multiple devices.
+
+By integrating passkeys, Sui improves security and accessibility, making it easier for users to manage their accounts without compromising decentralization or cryptographic security.
```
```suggestion
By supporting passkey signature scheme, Sui improves security and accessibility, making it easier for users to manage their accounts with hardened security. Passkey wallets are also tied to the origin, meaning they can’t be phished or used on a different site, which makes it a safer user wallet option. 
```

joyqvq commented on docs/content/concepts/cryptography/passkeys.mdx:
```diff
@@ -0,0 +1,60 @@
+
+By integrating passkeys, Sui improves security and accessibility, making it easier for users to manage their accounts without compromising decentralization or cryptographic security.
+
+Refer to [Typescript SDK support](https://sdk.mystenlabs.com/typescript/cryptography/passkey) on how to add passkey support to your application. Also feel free to refer to [SIP-9](https://github.com/sui-foundation/sips/blob/main/sips/sip-9.md) for product specification.
```
nit: standardize the passkey usage, referring to it as passkey instead of passkeys (to be consistent with code)

joyqvq commented on docs/content/concepts/cryptography/passkeys.mdx:
```diff
@@ -0,0 +1,60 @@
+
+Users can sign transactions directly on devices with built-in authenticators (such as Face ID on iPhones or Windows Hello on Windows PCs). This approach lets users sign transactions natively without needing an external security key.
+
+**Recover access and secure accounts with multi-signature authentication**
```
the cloud synchronization has nothing to do with multisig, just mention the passkey is recoverable if synced in cloud is sufficient.

joyqvq commented on docs/content/concepts/cryptography/passkeys.mdx:
```diff
@@ -0,0 +1,60 @@
+
+Users cannot transfer passkeys between different authenticators. For example, a passkey created on a security key cannot move to another device unless it syncs through a cloud provider. To avoid losing access, users should set up authentication on multiple devices.
+
+**Multisig support is not available yet**
```
we can mark this as available in testnet or remove mentioning this

joyqvq commented:
lgtm with inlined comments addressed

abhinavg6 commented:
Thanks @joyqvq - addressed all the feedback

@ronny-mysten - Can I request you to please review and merge by Monday? This would be deployed to testnet with Tuesday release. Thanks.

ronny-mysten commented on docs/content/concepts/cryptography/passkeys.mdx:
```diff
@@ -0,0 +1,56 @@
+
+By supporting the passkey signature scheme, Sui improves security and accessibility, making it easier for users to manage their accounts with hardened security. Passkey-based wallets are also tied to the origin, meaning they can’t be phished or used on a different site, which makes it a more secure authentication option.
+
+Refer to the [Typescript SDK support](https://sdk.mystenlabs.com/typescript/cryptography/passkey) on how to add passkey support to your application. Also feel free to refer to [SIP-9](https://github.com/sui-foundation/sips/blob/main/sips/sip-9.md) for product specification.
```
```suggestion
Refer to the [Typescript SDK support](https://sdk.mystenlabs.com/typescript/cryptography/passkey) on how to add passkey support to your application. For product specification, refer to [SIP-9](https://github.com/sui-foundation/sips/blob/main/sips/sip-9.md).
```

ronny-mysten commented on docs/content/concepts/cryptography/passkeys.mdx:
```diff
@@ -0,0 +1,56 @@
+
+**Passkey functionality varies by authenticator**
+
+Some security keys do not support biometric authentication, requiring users to enter a PIN instead. Also since WebAuthn does not provide access to private keys, users must store their passkey securely or enable cloud synchronization for recovery.
```
```suggestion
Some security keys do not support biometric authentication, requiring users to enter a PIN instead. Also, because WebAuthn does not provide access to private keys, users must store their passkey securely or enable cloud synchronization for recovery.
```
